nodebuilder/state/core.go

```diff
@@ -30,14 +33,31 @@ func coreAccessor(
 	*modfraud.ServiceBreaker[*state.CoreAccessor, *header.ExtendedHeader],
 	error,
 ) {
-	ca, err := state.NewCoreAccessor(keyring, string(keyname), sync, corecfg.IP, corecfg.GRPCPort,
-		network.String(), opts...)
+	if corecfg.TLSEnabled {
+		tlsCfg, err := core.TLS(corecfg.TLSPath)
+		switch {
+		case err == nil:
+		case errors.Is(err, os.ErrNotExist):
+			// set an empty config if path is empty under `TLSEnabled=true`
+			tlsCfg = core.EmptyTLSConfig()
+		default:
+			return nil, nil, nil, err
+		}
+
+		xtoken, err := core.XToken(corecfg.XTokenPath)
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return nil, nil, nil, err
+		}
+		opts = append(opts, state.WithTLSConfig(tlsCfg), state.WithXToken(xtoken))
```

Comment:
in this case xToken is `""` so, I don't think it will break any logic but from the other - less code fewer problems 🙂 . I can add a comment explaining the usage of `state.WithXToken`